### PR TITLE
IGNITE-18413 Reorganize metastorage modules

### PR DESCRIPTION
**Pull request by sashapolo**
https://issues.apache.org/jira/browse/IGNITE-18413

### CODE REVIEW

**Review by ibessonov — APPROVED** [review]
Let's wait for @agura's approval as well

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/OperationType.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * limitations under the License.
  */
 
-package org.apache.ignite.internal.metastorage.common;
+package org.apache.ignite.internal.metastorage.dsl;
```
Comment (agura) [inline modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/OperationType.java:18]: I really don't like the `dsl` name. There is still a difference between a language and an API. Really minor, but I can't be silent :)

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
+/*
```
[the remark this answers is not in the document]
Comment (sashapolo) [reply modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/MetaStorageManager.java:1]: I agree with you, however this is how it was implemented before my changes, I simply extracted an interface. I don't want to rename this in my PR in order to reduce the amount of changes.

File: modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/OperationType.java (hunk 1 above)
Comment (sashapolo) [reply modules/metastorage-api/src/main/java/org/apache/ignite/internal/metastorage/dsl/OperationType.java:18]: Sure, do you have a better name in mind?

**Review by agura — APPROVED** [review]

**Pull request merged**